crates/sui-adapter/src/adapter.rs

```diff
@@ -568,8 +568,13 @@ fn process_successful_execution<S: Storage + ParentSync>(
         let has_public_transfer = abilities.has_store();
         debug_assert_eq!(
             id,
-            ObjectID::try_from(&contents[0..ID_END_INDEX])
-                .expect("object contents should start with an id")
+            ObjectID::try_from(contents.get(0..ID_END_INDEX).ok_or_else(|| {
+                ExecutionError::new_with_source(
+                    ExecutionErrorKind::InvariantViolation,
+                    "Cannot parse Object ID",
+                )
+            })?)
+            .expect("object contents should start with an id")
```

Comment:
unrelated to the array access, but this expect here would trigger a panic here, do we want to this to be an execution error instead?

Reply:
My 2c on the issue: I want to offer  rule of thumb to decide whether to panic or error:
- Any condition that can be triggered by user/other validator needs to be an error and can not be panic, obviously
- Any internal invariant violation should be panic - this allows to surface invariants violations sooner and simplifies reasoning about the code by highlighting that this condition simply can't exist

Reply:
@andll can we mark which of these fixes in this PR conform to Andrey's categorization?
also please add docs for cases in which panic is accepted. Ie explain in a comment that this is critical and we should panic here.... so next devs who deal with that file will know

Reply:
In this case `contents.get(0..ID_END_INDEX).ok_or_else(|| {...})?` is infallible: it either returns an error or resolves to a correct expression. This seems to call for using `ObjectID::from_bytes` rather than `try_from`, and removing the expect.